### PR TITLE
tweak the reduce exercise for beginner friendliness

### PR DESCRIPTION
**Pull request by Marlena**
In working through the reduce example at the Oakland Node school today, my pair and I thought that this tweak to the suggested solution along with a helpful hint would make the reduce example a little bit easier to solve.

Let me know if there are any additional tweaks or changes I should make.

Thanks for making this workshopper. It has been really fun!

### CODE REVIEW

File: exercises/basic_reduce/solution/solution.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 function countWords(arr) {
-  return arr.reduce(function(countMap, word) {
-    countMap[word] = ++countMap[word] || 1 // increment or initialize to 1
-    return countMap
-  }, {}) // second argument to reduce initialises countMap to {}
+  return arr.reduce(function(accumulator, current_word) {
```
Comment (timoxley) [inline exercises/basic_reduce/solution/solution.js]: The original argument names were intended to be "semantic", related to the problem, rather than generic "accumulator". i.e. you're creating a `map` of `counts` for `words`.

Reduce functions can be hard to follow, so the idea is to embed as much semantic information into the argument names as possible. I'm open to better names, so long as they're not generic.

File: exercises/basic_reduce/problem.md [hunk 2]
```diff
@@ -40,3 +40,21 @@ function countWords(inputWords) {
...
 module.exports = countWords
 ```
+
+Helps & Hints: 
+The main idea of a reduce function is that you are reducing the values in an array down to a single value.
+  
+This single value is often called the accumulator because it is inside of this variable that values are accumulated.
+
+When you use the javascript array method for reduce, many of the examples have a one return for this method, but it can be easier
+to return the accumulator after performing your work.
```
Comment (timoxley) [inline exercises/basic_reduce/problem.md]: I'd just drop this paragraph. Better to just present one style and not mention others.

File: exercises/basic_reduce/problem.md [hunk 3]
```diff
@@ -40,3 +40,21 @@ function countWords(inputWords) {
...
+When you use the javascript array method for reduce, many of the examples have a one return for this method, but it can be easier
+to return the accumulator after performing your work. 
+
+```js
+var sum = [0, 1, 2, 3].reduce(function(accumulator, current_value) {
+  
+    accumulator = accumulator + current_value 
+  
+    return accumulator;
+}, 0);
```
Comment (timoxley) [inline exercises/basic_reduce/problem.md]: For consistency with other code samples, please drop unnecessary semicolons, unnecessary line breaks and use camelCase for variables. :D

**Review by timoxley — CHANGES_REQUESTED** [review]
Some minor changes needed but this looks good! Thanks.

Wish we had these explanations for filter & map too.

File: exercises/basic_reduce/solution/solution.js (hunk 1 above)
Comment (Marlena) [reply exercises/basic_reduce/solution/solution.js]: I see your point.  Thinking of it as an accumulator helped me understand reduce better, but using semantic naming is also helpful.
